### ISSUE TITLE
how to train instance segmentation using my dataset or load mature model such as unet of pytorch

### ISSUE THREAD

**Issue by yanchaoguo** [awslabs/djl#102]
Python's UNET model works very well. How can I load it into djl

**Comment by yanchaoguo** [comment]
打算通过图片语义分隔的方式进行背景移除（抠图）,目前demo中使用的模型准度不满足要求 ，边缘处理不够和谐

**Comment by lanking520** [comment]
As long as the model is [tracable](https://pytorch.org/docs/master/generated/torch.jit.trace.html) you can import in DJL and start using it. You may need to create your own translator. For the pretrained model DJL provided, it use the [GluonCV](https://gluon-cv.mxnet.io/build/examples_segmentation/demo_fcn.html) model which ideally should work well. The reason you saw the saw-tooth like segmentation is caused by our toy translator that only do the square mapping. In reality, you should try to implement something that matplotlib provided which would bring you better result.

**Comment by yanchaoguo** [comment]
> As long as the model is [tracable](https://pytorch.org/docs/master/generated/torch.jit.trace.html) you can import in DJL and start using it. You may need to create your own translator. For the pretrained model DJL provided, it use the [GluonCV](https://gluon-cv.mxnet.io/build/examples_segmentation/demo_fcn.html) model which ideally should work well. The reason you saw the saw-tooth like segmentation is caused by our toy translator that only do the square mapping. In reality, you should try to implement something that matplotlib provided which would bring you better result.

I don't quite understand how I should optimize to improve the sawtooth problem

**Comment by lanking520** [comment]
This is the [code to draw the mask](https://github.com/awslabs/djl/blob/master/api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java#L260). Essentially when we do instance segmentation, we obtained a Detected Object as well as a mask. Mask is a matrix contains distribution of region. We draw the result based on the region. There are advanced algorithm developed in Python to draw the mask into a better shape. The current Java solution is purely demo purpose and can be improve.

**Comment by stu1130** [comment]
@yanchaoguo if you have any other question, feel free to reopen the issue.

**Comment by paipeng** [comment]
Hello DJL,


Any example for using U-Net model in DJL?

What I can figure out is, how to set output type?

Regards,

Pai

**Comment by lanking520** [comment]
Checkout the stable diffusion model. There is a PyTorch Unet used in between for latent generation.

**Comment by paipeng** [comment]
Thank you,

CategoryMask works as the output type for predict.
